### PR TITLE
fix/TECH 438 small fixes

### PR DESCRIPTION
**Pull request by SamTheisens**
- [TECH-438] Move toml to main dependencies
- [TECH-438] Fix dagster example
- Fix getting changes from merge commits

[TECH-438]: https://vandebron.atlassian.net/browse/TECH-438?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TECH-438]: https://vandebron.atlassian.net/browse/TECH-438?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
----
📕 [TECH-438](https://vandebron.atlassian.net/browse/TECH-438) Usability issues with MPyL sam@vandebron.nl 
[p1685111170053379](https://vandebron.slack.com/archives/C04KP2UFMMJ/p1685111170053379) 

🏗️ Build [41](https://jenkins.k8s-serv-backend.vdbinfra.nl/job/MPyL%20Pipeline%20-%20Test/job/PR-126/41/display/redirect) ✅ Successful, started by _Daniel Koves_  
🚀  *[cloudfront-service](https://cloudfront-service-126.test.nl/)*, *enrichChargeSessionsJob*, *job*, *kong-sync*, *[nodeservice](https://nodeservice-126.test.nl/)*, *[sbtservice](https://sbtservice-126.test.nl/)*

### CODE REVIEW

**Review by Jorg88 — APPROVED** [review]

File: src/mpyl/reporting/targets/jira.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def from_issue_response(response):
         ticket_id = response['key']
         assignee = fields.get('assignee')
         user = assignee or fields.get('reporter') or fields.get('creator')
-        avatar = user['avatarUrls']['24x24'] if user else ''
+        avatar = user['avatarUrls']['24x24'] if user and user['avatarUrls']['24x24'].endswith('24') else ''
```
Comment (koeves) [inline src/mpyl/reporting/targets/jira.py:61]: We can also try to set the avatar in `to_markdown_summary` with an HTML tag 
`f''`

**Comment by koeves** [comment]
@SamTheisens is this ready to be merged?

**Comment by SamTheisens** [comment]
Yes, @koeves, if it looks good to you and works well with the monorepo (which I can't test). 👍

File: src/mpyl/steps/deploy/kubernetes.py [hunk 2]
```diff
@@ -53,6 +55,16 @@ def execute(self, step_input: Input) -> Output:
                 endpoint = '/' if has_specific_routes_configured else '/swagger/index.html'
                 spec[DEPLOYED_SERVICE_KEY] = f'{hostname}{endpoint}'
             artifact = input_to_artifact(ArtifactType.DEPLOYED_HELM_APP, step_input, spec=spec)
+            if properties.target == Target.PRODUCTION:
```
Comment (Jorg88) [inline src/mpyl/steps/deploy/kubernetes.py:58]: @koeves how do we test this?

**Review by Jorg88 — APPROVED** [review]

File: src/mpyl/steps/deploy/kubernetes.py (hunk 2 above)
Comment (koeves) [reply src/mpyl/steps/deploy/kubernetes.py:58]: I will try to find a project that I can safely mess with in production (we have a mock service for example). Then I can check if the same image is deployed to test

File: src/mpyl/stages/discovery.py [hunk 3]
```diff
@@ -66,9 +66,9 @@ def find_invalidated_projects_for_stage(all_projects: set[Project], stage: Stage
     return set(filter(lambda p: are_invalidated(p, stage, change_history), all_projects))
 
 
-def find_deploy_set(repo_config: RepoConfig) -> DeploySet:
+def find_deploy_set(repo_config: RepoConfig, tag: Optional[str]) -> DeploySet:
     with Repository(repo_config) as repo:
-        changes_in_branch = repo.changes_in_branch_including_local()
+        changes_in_branch = repo.changes_in_branch_including_local() if not tag else repo.changes_in_tagged_commit(tag)
```
Comment (Jorg88) [inline src/mpyl/stages/discovery.py:71]:
```suggestion
        changes_in_branch = repo.changes_in_tagged_commit(tag) if tag else repo.changes_in_branch_including_local()
```

super minor, but I've learned that checking for 'true' is always easier to read

**Review by Jorg88 — APPROVED** [review]

**Pull request merged**